File: Data Structures & Algorithms/reverse-integer/submission-0.java
class Solution {
    public int reverse(int x) {
        int max = Integer.MAX_VALUE;
        int min = Integer.MIN_VALUE;
        int res = 0;
        while(x!=0){
            int rem = x%10;
            x /=10;
            if(res>max/10||(res==max/10 && rem  > max%10)) return 0;
            if(res<min/10||(res==min/10 && rem< min%10)) return 0;
            res = res*10 + rem;
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        freq={}
        repeAlph=0
        l=0
        res=0
        for r in range(len(s)):
            freq[s[r]]=1+freq.get(s[r],0)
            repeAlph=max(repeAlph,freq[s[r]])
            
            while (r-l+1) - repeAlph>k:
                freq[s[l]]-=1
                l+=1
            res=max(res,r-l+1)
        return res
